paddle/operators/demo_op.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "paddle/framework/op_registry.h"
+
+using namespace paddle::framework;
+
+namespace paddle {
+namespace operators {
+
+class CosineOp : public OperatorBase {
+ public:
+  void Run(OpContext *context) const override {
+    printf("%s\n", DebugString().c_str());
+  }
+};
+
+class CosineOpProtoAndCheckerMaker : public OpProtoAndCheckerMaker {
+ public:
+  CosineOpProtoAndCheckerMaker(OpProto *proto, OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("input", "input of cosine op");
+    AddOutput("output", "output of cosine op");
+    AddAttr<float>("scale", "scale of cosine op")
+        .SetDefault(1.0)
+        .LargerThan(0.0);
+    AddType("cos");
+    AddComment("This is cos op");
+  }
+};
+
+REGISTER_OP(CosineOp, CosineOpProtoAndCheckerMaker, cos_sim)
```

Comment:
yes ,the registry for different device type(CPU/GPU) and data type(float/int/float16) should be optimized later.